doc/mgmt/SONiC_OC_Vlan_HLD.md

```diff
@@ -0,0 +1,302 @@
+2. [openconfig-if-ethernet.yang](https://github.com/openconfig/public/blob/master/release/models/interfaces/openconfig-if-ethernet.yang)
+3. [sonic-vlan.yang](https://github.com/project-arlo/sonic-mgmt-framework/blob/master/models/yang/sonic/sonic-vlan.yang)
+
+Supported yang objects and attributes are highlighted in green:
+```diff
+module: openconfig-interfaces
++   +--rw interfaces
++      +--rw interface* [name]
++         +--rw name                   -> ../config/name
+          +--rw config
+          |  +--rw name?            string
+          |  +--rw type             identityref
++         |  +--rw mtu?             uint16
+          |  +--rw loopback-mode?   boolean
+          |  +--rw description?     string
++         |  +--rw enabled?         boolean
+          |  +--rw oc-vlan:tpid?    identityref
+          +--ro state
+          |  +--ro name?            string
+          |  +--ro type             identityref
++         |  +--ro mtu?             uint16
+          |  +--ro loopback-mode?   boolean
+          |  +--ro description?     string
+          |  +--ro enabled?         boolean
+          |  +--ro ifindex?         uint32
++         |  +--ro admin-status     enumeration
++         |  +--ro oper-status      enumeration
+          |  +--ro last-change?     oc-types:timeticks64
+          |  +--ro logical?         boolean
+          |  +--ro oc-vlan:tpid?    identityref
+          +--rw hold-time
+          |  +--rw config
+          |  |  +--rw up?     uint32
+          |  |  +--rw down?   uint32
+          |  +--ro state
+          |     +--ro up?     uint32
+          |     +--ro down?   uint32
+          +--rw oc-eth:ethernet
++         |  +--rw oc-vlan:switched-vlan
++         |     +--rw oc-vlan:config
++         |     |  +--rw oc-vlan:interface-mode?   oc-vlan-types:vlan-mode-type
+          |     |  +--rw oc-vlan:native-vlan?      oc-vlan-types:vlan-id
++         |     |  +--rw oc-vlan:access-vlan?      oc-vlan-types:vlan-id
++         |     |  +--rw oc-vlan:trunk-vlans*      union
++         |     +--ro oc-vlan:state
+          |        +--ro oc-vlan:interface-mode?   oc-vlan-types:vlan-mode-type
+          |        +--ro oc-vlan:native-vlan?      oc-vlan-types:vlan-id
++         |        +--ro oc-vlan:access-vlan?      oc-vlan-types:vlan-id
++         |        +--ro oc-vlan:trunk-vlans*      union
+```
+### 3.2.2 CLI
+- sonic-vlan.yang is used for CLI #show commands.
+
+#### 3.2.2.1 Configuration Commands
+
+#### VLAN Creation
+`interface Vlan <vlan-id>`
+```
+sonic(config)# interface Vlan 5
+```
+#### VLAN Deletion
+`no interface Vlan <vlan-id>`
+```
+sonic(config)# no interface Vlan 5
+```
+#### MTU Configuration
+`mtu <mtu-val>`
+```
+sonic(conf-if-vlan20)# mtu 2500
+sonic(conf-if-vlan20)# no mtu
+```
+#### MTU Removal
+`no mtu <mtu-val>` --> Reset to default
+```
+sonic(conf-if-vlan20)# no mtu
+```
+#### Disable VLAN
+`shutdown`
+```
+sonic(conf-if-vlan20)# shutdown
+```
+#### Enable VLAN
+`no shutdown`
+```
```

Comment:
sonic doesn't have these support. please hide these commands.

Reply:
okay sure

Reply:
If IP address config is coming into this interface, then this may be required.  with this shutdown config,  VLAN IP interface will be shutdown.

Reply:
Based on discussion with @ben-gale & others in team we won't be supporting shutdown command for Buzznik. @justinejose91 - To remove shutdown command from HLD. .